src/Lexer.js

```diff
@@ -73,6 +73,22 @@ module.exports = class Lexer {
       }
     }
     this.tokenizer.rules = rules;
+
+    this.blockTokenizers = [
+      this.newline,
```

Comment:
We need some way to get the name of the function so we can add new functions before and after certain ones.

Maybe something like:

```suggestion
      { name: "newline", func: this.newline},
```

Reply:
Ok. What do you think about using a [Map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of an Array of one-element objects? It requires es6 but keeps insertion order for iterability. I have no idea about speed.

Reply:
Sounds good. We won't support ie11 by the time this gets merged anyway.

Reply:
> We won't support ie11 by the time this gets merged anyway.

When are we expecting to merge this? Are we planning to do a major semver soon?

Reply:
Ya we have to bump the major version for #1864 anyway I figured we could add a few other breaking changes. The [v2](https://github.com/markedjs/marked/projects/7) project has a few things we could get in to v2. I'm not sure if we will get all of them in but I figured we could seriously look into them and if they are viable in a reasonable amount of time we could release v2 in a month or two.